models/product-data/src/transformers.ts

```diff
@@ -0,0 +1,103 @@
+import { CategoryReference } from '@commercetools/platform-sdk';
+import { LocalizedString } from '@commercetools-test-data/commons';
+import {
+  buildField,
+  buildFields,
+  Transformer,
+} from '@commercetools-test-data/core';
+import type {
+  TCategoryOrderHintGraphql,
+  TProductData,
+  TProductDataGraphql,
+  TProductDataRest,
+} from './types';
+
+const transformers = {
+  default: Transformer<TProductData, TProductData>('default', {
+    buildFields: [
+      'name',
+      'categories',
+      'description',
+      'slug',
+      'metaTitle',
+      'metaDescription',
+      'metaKeywords',
+      'masterVariant',
+      'variant',
+      'variants',
+      'allVariants',
+      'searchKeywords',
+      'searchKeyword',
+      'categoriesRef',
+    ],
+  }),
+  rest: Transformer<TProductData, TProductDataRest>('rest', {
+    replaceFields: ({ fields }) => {
+      const { categories } = fields;
+
+      const categoryReferences: Array<CategoryReference> = buildFields(
+        categories
+      ).map((category) => ({
+        id: category.id,
+        typeId: 'category',
+      }));
+
+      /**
+       * We cannot use `replaceFields` in conjunction with `buildFields`,
+       * so we need to explicity build fields where necessary.
+       */
```

Comment:
@emmenko Just wanted to make sure this is expected behavior

Reply:
In theory building fields can always be done. See implementation https://github.com/commercetools/test-data/blob/e74a09a2a55e258d7c63cf626d58c7e4978ca3c9/core/src/transformer.ts#L20-L39

Reply:
Ah I think we need to pass here the `transformedFields`

https://github.com/commercetools/test-data/blob/e74a09a2a55e258d7c63cf626d58c7e4978ca3c9/core/src/transformer.ts#L60

We can look into that separately. Thanks for bringing it up.

Reply:
Or feel free to try it out here first

Reply:
This tracks with some tweaking I did on my branch. I was able to pass `fieldsReplacer({ fields: transformedFields })` which would give the replacer access to the built fields 👍 

I can make that change as well. Existing tests pass so it seems like it wouldn't cause regression

Reply:
@emmenko Resolved in 01702640abe122afd53c66a6578f26f32a6243b9